docs/source/en/core/development.md

```diff
@@ -1 +1,339 @@
+
+### Ignore Files
+
+To ignore some files in code coverage rate calculation, you may use `COV_EXCLUDES` environment variable to achieve this:
+
+```bash
+$ COV_EXCLUDES=app/plugins/c* npm run cov
+$ # Or pass this setting via parameters
+$ npm run cov -- --x=app/plugins/c*
+```
+
+## Debug
+
+### Print Logs
+
+### Use Logger Module
+
+There's built-in [Log](./logger.md) in egg. You may use `logger.debug()` to print out debug information. **It is recommended to be used in your implementation.**
+
+```js
+// controller
+this.logger.debug('current user: %j', this.user);
+
+// service
+this.ctx.logger.debug('debug info from service');
+
+// app/init.js
+app.logger.debug('app init');
+```
+
+Levels of logs can be configured via `config.logger.level` for printing into file, and `config.logger.consoleLevel` for printing into console.
+
+### Use Debug Module
+
+[debug](https://www.npmjs.com/package/debug) module is a debug tool which is widely adopted in Node.js community. Plenty of modules are using it to print debug information. Egg community has widely adopted it as well. **It is recommended to be used in framework and plugin development**
+
+To watch on the running process, we may start with certain code for debugging via `DEBUG` enviroment variable.
+
+(Do not confuse debug module and logger module. There are lots more stuff provided by logger module, but the log here we are talking about is debug information.)
+
+Turn on log of all modules:
+
+```bash
+$ DEBUG=* npm run dev
+```
+
+Turn on log of specific module:
+
+```bash
+$ DEBUG=egg* npm run dev
+```
+
+Detail logs of unit testing progress are able to be viewed via `DEBUG=* npm test`.
+
+### Debug With egg-bin
+
+#### Add Script
+
+Add `npm scripts` into `package.json`：
+
+```json
+{
+  "scripts": {
+    "debug": "egg-bin debug"
+  }
+}
+```
+
+And then we may set breakpoints for debugging our app via `npm run debug`.
+
+`egg-bin` will select debug protocol automatically. [Inspector Protocol] will be selected for version 8.x or higher. For version lower than 8.x, it will select [Legacy Protocol].
+
+In the meantime, it supports custom debug parameters.
+
+```bash
+$ egg-bin debug --inpsect=9229
+```
+
+- Debug port of `master` is 9229 or 5858 (Legacy Protocol).
+- Debug port of `agent` is fixed to 5800, it is customizable via `process.env.EGG_AGENT_DEBUG_PORT`.
+- Debug port number of `worker` will increase from `master` port number.
+- During developing period, worker will be hot reloaded once code changed, which will cause debug port to be self incrementing. Please refer to the following IDE configuration for auto-reconnecting.
+
+#### Config Environment
+
+App is also started by `env: local` when executing `debug` command. The configuration being read is an object combined from `config.local.js` and `config.unittest.js` files.
```

Comment:
App is also started by `env: local` when executing `debug` . The configuration comes from the combination of both `config.local.js` and `config.unittest.js`.